force-app/main/default/triggers/OpportunityTrigger.trigger

```diff
@@ -0,0 +1,43 @@
+trigger OpportunityTrigger on Opportunity (before update, before delete) {
+  if (Trigger.isBefore) {
+
+    if (Trigger.isUpdate) {
+
+      List<Contact> contactsCEO = [SELECT Id, AccountId FROM Contact WHERE Title = 'CEO'];
+
+      Map<Id,Id> contactIdToAccountId = new Map<Id,Id>();
+
+      for (Contact contact : contactsCEO) {
+        contactIdToAccountId.put(contact.AccountId, contact.Id);
+      }
+
+
+      for (Opportunity opp : Trigger.new) {
+
+  //Question 5
+        if (opp.Amount <= 5000 ) {
+          opp.Amount.addError('Opportunity amount must be greater than 5000');
+        }
+
+  //Question 7
+        if (opp.AccountId != Null && contactIdToAccountId.get(opp.AccountId) != Null && opp.Primary_Contact__c != contactIdToAccountId.get(opp.AccountId)) {
+          opp.Primary_Contact__c = contactIdToAccountId.get(opp.AccountId);
+        }
+
+      }
+    }
+    if (Trigger.isDelete) {
+
+  //Question 6
+      List<Account> bankingAccount = [SELECT Id FROM Account WHERE Industry = 'Banking'];
```

Comment:
This is the same problem as above. This queries way more data than you need and may cause limit issues.